File: page/compliance_centre/compliance_centre.js
frappe.pages['compliance-centre'].on_page_load = function (wrapper) {

	const page = frappe.ui.make_app_page({
		parent: wrapper,
		title: 'Compliance Centre',
		single_column: true
	});

	const app = $(`
		<div class="p-4">

			<div id="filters"></div>

			<div class="mt-4">
				<button class="btn btn-primary" id="generate_btn">
					Generate
				</button>
			</div>

		</div>
	`);

	$(page.body).append(app);

	// Filters will be added here.

	$('#generate_btn').on('click', function () {

		// Collect filters

		// Call backend

		// Download PDF

	});

};

File: page/compliance_centre/compliance_centre.py
import frappe


@frappe.whitelist()
def generate(register, filters=None):
    """
    Compliance Centre launcher.

    Receives:
        register
        filters

    Calls the selected register generator.

    Returns:
        PDF
    """

    if register == "Wage Register":
        from red_statutory.registers.wage_register.pdf import generate
        return generate(filters)

    frappe.throw("Unsupported Register")